### PR TITLE
Always load netCDF metadata

### LINKED ISSUE #137: Passing missing data values to constructor fails with S3 storage backend
For example:
```
uri = "s3://bucket/object.nc"
Active(uri, "ta", "s3", missing_value=999.)
```

This uses a different code path which avoids loading the netCDF file metadata. Since #132 merged, this no longer works when using the S3 storage backend because we need to get the dtype from the metadata. It fails in the following way:

```
AttributeError: 'Active' object has no attribute '_dtype'
```

A simple solution is to always load the netCDF metadata. This has the downside that it is not necessary when using local storage and missing data has been specified.

### PR DESCRIPTION
**Pull request by markgoddard**
Since byte order support was added in #132, we capture the dtype from
netCDF metadata and use it in preference to the Zarr dtype when using
the S3 storage backend.

In the case where missing data values are passed into the Active
constructor, we do not load netCDF metadata, and therefore do not
capture the dtype. This leads to the following error with S3 storage:

  AttributeError: 'Active' object has no attribute '_dtype'

This change switches to always load the metadata so that we can capture
the dtype.  This has the downside that it is not necessary when using
local storage and missing data has been specified.

This allows us to enable the test_daily_data_masked test for S3.

Closes #137

### CODE REVIEW

**Comment by markgoddard** [comment]
Please wait for #136 to merge before merging this PR.

File: activestorage/active.py
Comment (valeriupredoi) [inline activestorage/active.py:111]: yes, always (even asked @davidhassell and @bnlawrence just to be 120% sure). BTW pls don't use `/` in branch names, bud - it confuses the heck out of my local `refs`

**Review by valeriupredoi — APPROVED** [review]
good call and solid fix! :beer:

**Pull request merged**
